diff --git a/configure b/configure
index 34c2adb4a4..683945effc 100755
--- a/configure
+++ b/configure
@@ -3713,11 +3713,11 @@ LIBNAME='$(LIBPREF)$(FULLNAME)$(LIBSUF)'
 SLIBPREF="lib"
 SLIBSUF=".so"
 SLIBNAME='$(SLIBPREF)$(FULLNAME)$(SLIBSUF)'
-SLIBNAME_WITH_VERSION='$(SLIBNAME).$(LIBVERSION)'
-SLIBNAME_WITH_MAJOR='$(SLIBNAME).$(LIBMAJOR)'
+SLIBNAME_WITH_VERSION='$(SLIBNAME)'
+SLIBNAME_WITH_MAJOR='$(SLIBNAME)'
 LIB_INSTALL_EXTRA_CMD='$$(RANLIB) "$(LIBDIR)/$(LIBNAME)"'
-SLIB_INSTALL_NAME='$(SLIBNAME_WITH_VERSION)'
-SLIB_INSTALL_LINKS='$(SLIBNAME_WITH_MAJOR) $(SLIBNAME)'
+SLIB_INSTALL_NAME='$(SLIBNAME)'
+SLIB_INSTALL_LINKS=
 VERSION_SCRIPT_POSTPROCESS_CMD="cat"
 
 asflags_filter=echo
@@ -5220,7 +5220,7 @@ case $target_os in
         add_ldexeflags -fPIE -pie
         SLIB_INSTALL_NAME='$(SLIBNAME)'
         SLIB_INSTALL_LINKS=
-        SHFLAGS='-shared -Wl,-soname,$(SLIBNAME)'
+        SHFLAGS='-shared -soname $(SLIBNAME)'
         ;;
     haiku)
         prefix_default="/boot/common"
